### PR TITLE
Fix BC issue on AclMatrixType

### PR DESCRIPTION
**Pull request by turko**
[FUP] https://github.com/sonata-project/SonataAdminBundle/pull/3746

Thank for the feedback. Good to know. 

Includes todo when dropping sf < 2.8.

### CODE REVIEW

File: Util/AdminObjectAclManipulator.php [hunk 1]
```diff
@@ -271,8 +271,21 @@ protected function buildForm(AdminObjectAclData $data, FormBuilderInterface $for
                     'attr'     => $attr,
                 );
             }
-
-            $formBuilder->add($key, new AclMatrixType(), array('permissions' => $permissions, 'acl_value' => $aclValue));
+            //@todo Remove when dropping Symfony <2.8 support
+            $aclMatrixType = new AclMatrixType();
```
Comment (Soullivaneuh) [inline Util/AdminObjectAclManipulator.php]: This is not the good method. Here, you are instantiating a class sometime for nothing.

Please use Symfony `AbstractType` FQDN on `method_exists` instead.

File: Util/AdminObjectAclManipulator.php [hunk 2]
```diff
@@ -271,8 +271,21 @@ protected function buildForm(AdminObjectAclData $data, FormBuilderInterface $for
...
+            if (method_exists($aclMatrixType, 'getBlockPrefix')) {
+                $formBuilder->add(
+                    $key,
+                    'Sonata\AdminBundle\Form\Type\AclMatrixType',
```
Comment (Soullivaneuh) [inline Util/AdminObjectAclManipulator.php]: The BC trick should be only here, with a ternary condition for example.

Not all the FormBuilder stuff should be put here.

**Comment by Soullivaneuh** [comment]
Please remove unrelated commits and change your commit comment by the title I put on this PR.

**Comment by OskarStark** [comment]
ping @turko

**Comment by core23** [comment]
Ping @turko

**Comment by core23** [comment]
Closing in favor of #4084

**Pull request closed without merging**